python/tests/core/metrics/test_condition_count.py

```diff
@@ -0,0 +1,61 @@
+import re
+from typing import Dict
+
+from whylogs.core.dataset_profile import DatasetProfile
+from whylogs.core.datatypes import DataType
+from whylogs.core.metrics import Metric
+from whylogs.core.metrics.condition_count_metric import (
+    ConditionCountConfig,
+    ConditionCountMetric,
+)
+from whylogs.core.preprocessing import PreprocessedColumn
+from whylogs.core.resolvers import Resolver
+from whylogs.core.schema import ColumnSchema, DatasetSchema
+
+
+def test_condition_count_metric() -> None:
+    conditions = {
+        "alpha": re.compile("[a-zA-Z]+"),
+        "digit": re.compile("[0-9]+"),
+    }
+    metric = ConditionCountMetric(conditions)
+    strings = ["abc", "123", "kwatz", "314159", "abc123"]
+    metric.columnar_update(PreprocessedColumn.apply(strings))
+    summary = metric.to_summary_dict(None)
+
+    assert set(summary.keys()) == {"total", "alpha", "digit"}
+    assert summary["total"] == len(strings)
+    assert summary["alpha"] == 3  # "abc123" matches since it's not fullmatch
+    assert summary["digit"] == 2
+
+
+def test_condition_count_in_profile() -> None:
+    class TestResolver(Resolver):
+        def resolve(self, name: str, why_type: DataType, column_schema: ColumnSchema) -> Dict[str, Metric]:
+            return {"condition_count": ConditionCountMetric.zero(column_schema.cfg)}
+
+    conditions = {
+        "alpha": re.compile("[a-zA-Z]+"),
+        "digit": re.compile("[0-9]+"),
+    }
+    config = ConditionCountConfig(conditions)
+    resolver = TestResolver()
+    schema = DatasetSchema(default_configs=config, resolvers=resolver)
```

Comment:
Lots of setup for a non-standard metrics :(  

I'm thinking about adding API support for dynamically adding metrics & conditions.